iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -91,6 +110,65 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
             }
         }
 
+        private async Task GetWritablePropertiesAndHandleChangesAsync()
+        {
+            ClientProperties properties = await _deviceClient.GetClientPropertiesAsync();
+            ClientPropertyCollection writableProperties = properties.WritablePropertyRequests;
+            long serverWritablePropertiesVersion = writableProperties.Version;
+
+            // Check if the writable property version is outdated on the local side.
+            // For the purpose of this sample, we'll only check the writable property versions between local and server
+            // side without comparing the property values.
+            if (serverWritablePropertiesVersion > s_localWritablePropertiesVersion)
+            {
+                _logger.LogDebug($"The writable property version cached on local is changing from {s_localWritablePropertiesVersion} to {serverWritablePropertiesVersion}.");
+
+                foreach (KeyValuePair<string, object> writableProperty in writableProperties)
+                {
+                    switch (writableProperty.Key)
+                    {
+                        case Thermostat1:
+                        case Thermostat2:
+                            const string targetTemperatureProperty = "targetTemperature";
+                            string componentName = writableProperty.Key;
+
+                            if (writableProperties.TryGetValue(componentName, targetTemperatureProperty, out double targetTemperatureValue))
+                            { 
+                                _logger.LogDebug($"Property: Received - component=\"{componentName}\", [ \"{targetTemperatureProperty}\": {targetTemperatureValue}°C ].");
+
+                                _temperature[componentName] = targetTemperatureValue;
+
+                                var propertyValue = SystemTextJsonPayloadSerializer.Instance.CreateWritablePropertyResponse(_temperature[componentName], CommonClientResponseCodes.OK, serverWritablePropertiesVersion);

```

Comment:
nit: this is a getting a bit too long; you might want to split the args into new lines.